lib/tedious.js

```diff
@@ -232,6 +232,14 @@ class ConnectionPool extends base.ConnectionPool {
       debug('pool(%d): connection #%d created', IDS.get(this), IDS.get(tedious))
       debug('connection(%d): establishing', IDS.get(tedious))
 
+      tedious.on('debug', msg => {
```

Comment:
is this needed?

Reply:
It was, because some disconnect events were not coming through, but I think this is one thing that in knex side was implemented in a different way in the patch I made there.

Reply:
So... probably this is not needed, but something else was.

Reply:
Yeah, to fix the pool depletion bug you needed the `safeReject` and `safeResolve` methods and to attach to the `end` event as well as the `error` because `end` was being fired by tedious (instead of error) if there was a socket timeout during connections.

That's not the case anymore so I've completely removed that patch, so this can go as well, I think.